src/keys-builder/template/structural-directive.extractor.ts

```diff
@@ -74,19 +76,35 @@ export function traverse(
   }
 }
 
-function unwrapExpression(exp: AST): AST {
-  /*
-   * Handle pipes in interpolation
-   *
-   * @example
-   * {{ t('another(test)') | pipe1 | pipe2 }} will return the `t('another(test)')` node
-   * */
-  return isBindingPipe(exp) ? unwrapExpression(exp.exp) : exp;
+/**
+ * extract method calls from an AST.
+ *
+ * @example
+ * {{ t('another(test)') | pipe1 | pipe2 }} will return the `t('another(test)')` node.
+ *
+ * @example
+ * {{ data | pipe: t('key') }} will return the `t('key')` node.
+ */
+function unwrapExpression(exp: AST): AST[] {
+  if (isBindingPipe(exp)) {
+    return [
+      ...unwrapExpression(exp.exp),
+      ...exp.args.map((arg) => unwrapExpression(arg)).flat(),
+    ];
+  } else if (isLiteralMap(exp)) {
+    return exp.values.map((value) => unwrapExpression(value)).flat();
+  }
+
+  return [exp];
 }
 
-function getMethodUsages(expressions: AST[], containers: ContainerMetaData[]) {
+function getMethodUsages(
+  expressions: AST[],
+  containers: ContainerMetaData[]
+): ContainerMetaData[] {
   return expressions
     .map((exp) => unwrapExpression(exp))
+    .flat()
```

Comment:
I would expect unwrapExpression to return a flat array, and seems like sometimes it does and sometimes it doesn't

Reply:
I refactored *unwrapExpression* so that it always returns a flat array.
We won't be able to remove that `flat()` because there is a `map` before it. We can use flatMap if you want to.

Reply:
Please use `flatMap`.